bin/network-monitor/src/status.rs

```diff
@@ -468,3 +485,265 @@ pub(crate) async fn check_remote_prover_status(
         },
     }
 }
+
+// EXPLORER STATUS CHECKER
+// ================================================================================================
+
+const LATEST_BLOCK_QUERY: &str = "
+query LatestBlock {
+    blocks(input: { sort_by: timestamp, order_by: desc }, first: 1) {
+        edges {
+            node {
+                block_number
+                timestamp
+                number_of_transactions
+                number_of_nullifiers
+                number_of_notes
+                block_commitment
+                chain_commitment
+                proof_commitment
+                number_of_account_updates
+            }
+        }
+    }
+}
+";
+
+#[derive(Serialize)]
+struct GraphqlRequest<V> {
+    query: &'static str,
+    variables: V,
+}
+
+fn latest_block_request() -> GraphqlRequest<serde_json::Value> {
+    GraphqlRequest {
+        query: LATEST_BLOCK_QUERY,
+        variables: serde_json::json!({}),
+    }
+}
+
+/// Runs a task that continuously checks explorer status and updates a watch channel.
+///
+/// This function spawns a task that periodically checks the explorer service status
+/// and sends updates through a watch channel.
+///
+/// # Arguments
+///
+/// * `explorer_url` - The URL of the explorer service.
+/// * `name` - The name of the explorer.
+/// * `status_sender` - The sender for the watch channel.
+/// * `status_check_interval` - The interval at which to check the status of the services.
+///
+/// # Returns
+///
+/// `Ok(())` if the monitoring task runs and completes successfully, or an error if there are
+/// connection issues or failures while checking the explorer status.
+#[instrument(target = COMPONENT, name = "explorer-status-task", skip_all)]
+pub async fn run_explorer_status_task(
+    explorer_url: Url,
+    name: String,
+    status_sender: watch::Sender<ServiceStatus>,
+    status_check_interval: Duration,
+    request_timeout: Duration,
+) {
+    let mut explorer_client = reqwest::Client::new();
+
+    let mut interval = tokio::time::interval(status_check_interval);
+    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+
+    loop {
+        interval.tick().await;
+
+        let current_time = current_unix_timestamp_secs();
+
+        let status = check_explorer_status(
+            &mut explorer_client,
+            explorer_url.clone(),
+            name.clone(),
+            current_time,
+            request_timeout,
+        )
+        .await;
+
+        // Send the status update; exit if no receivers (shutdown signal)
+        if status_sender.send(status).is_err() {
+            println!("No receivers for explorer status updates, shutting down");
+            info!("No receivers for explorer status updates, shutting down");
```

Comment:
nit (feel free to disregard): I'd remove the `println`; the info trace should be enough for the user